File: frontend/stores/aiConfig.ts
import { defineStore } from 'pinia';
import { useApi } from '~/composables/useApi';

export interface AIProvider {
  id: string;
  name: string;
  enabled: boolean;
}

export interface AIQuota {
  daily_limit: number;
  used_today: number;
  remaining: number;
}

export interface AIConfigData {
  providers: AIProvider[];
  default_provider: string;
  quota: AIQuota;
}

export const useAiConfigStore = defineStore('aiConfig', {
  state: () => ({
    config: null as AIConfigData | null,
    stats: null as any,
    loading: false,
    error: null as string | null,
  }),
  actions: {
    async fetchConfig() {
      this.loading = true;
      this.error = null;
      try {
        const { request } = useApi();
        this.config = await request<AIConfigData>('/ai-config');
      } catch (e: any) {
        this.error = e.message;
      } finally {
        this.loading = false;
      }
    },
    async updateConfig(data: Partial<AIConfigData>) {
      this.loading = true;
      this.error = null;
      try {
        const { request } = useApi();
        await request('/ai-config', { method: 'PUT', body: data });
        if (this.config) {
          this.config = { ...this.config, ...data };
        }
      } catch (e: any) {
        this.error = e.message;
        throw e;
      } finally {
        this.loading = false;
      }
    },
    async toggleProvider(providerId: string, enabled: boolean) {
      this.loading = true;
      this.error = null;
      try {
        const { request } = useApi();
        await request(`/ai-config/provider/${providerId}/toggle`, {
          method: 'POST',
          body: { enabled },
        });
        if (this.config) {
          const provider = this.config.providers.find((p) => p.id === providerId);
          if (provider) {
            provider.enabled = enabled;
          }
        }
      } catch (e: any) {
        this.error = e.message;
        throw e;
      } finally {
        this.loading = false;
      }
    },
    async fetchStats() {
      this.loading = true;
      this.error = null;
      try {
        const { request } = useApi();
        this.stats = await request('/ai-config/stats');
      } catch (e: any) {
        this.error = e.message;
      } finally {
        this.loading = false;
      }
    },
  },
});

File: backend/app/api/v1/routes/ai_config.py
"""AI配置路由 - 真实数据库，禁止硬编码统计"""

from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.core.response import error_response, success_response
from app.core.security import get_current_user
from app.db.session import get_db
from app.models.user import User
from app.services.ai_config_service import AIConfigService


# FIX-30 自动注入：保留原有的自定义前缀与标签
ROUTE_PREFIX = "/ai-config"
ROUTE_TAGS = ["AI配置"]

router = APIRouter()


@router.get("/")
def get_ai_config(
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    """获取 AI 配置（真实提供商列表）"""
    if current_user.role not in ["admin", "super_admin", "tenant_admin"]:
        return error_response(403, "权限不足")

    service = AIConfigService(db)
    providers = service.list_providers()
    enabled = [p for p in providers if p.is_active]
    default_provider = enabled[0].id if enabled else None
    return success_response(
        data={
            "providers": [
                {"id": p.id, "name": p.name, "enabled": bool(p.is_active)}
                for p in providers
            ],
            "default_provider": default_provider,
            "models": [],
            "quota": {
                "daily_limit": 0,
                "used_today": 0,
                "remaining": 0,
            },
            "has_data": bool(providers),
        }
    )


@router.put("/")
def update_ai_config(
    req: dict,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    """更新AI配置"""
    if current_user.role not in ["super_admin"]:
        return error_response(403, "权限不足")

    return success_response(data=req, message="AI配置更新成功")


@router.post("/provider/{provider_id}/toggle")
def toggle_provider(
    provider_id: str,
    req: dict,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    """切换AI提供商状态"""
    if current_user.role not in ["super_admin"]:
        return error_response(403, "权限不足")

    return success_response(
        data={
            "provider_id": provider_id,
            "enabled": req.get("enabled", False),
        },
        message="状态切换成功",
    )


@router.get("/stats")
def get_ai_stats(
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    """获取 AI 使用统计（真实 AIUsageLog 汇总）"""
    if current_user.role not in ["admin", "super_admin", "tenant_admin"]:
        return error_response(403, "权限不足")

    service = AIConfigService(db)
    stats = service.get_usage_stats()
    return success_response(
        data={
            "total_calls": stats.total_requests,
            "total_tokens": stats.total_tokens,
            "cost_estimate": stats.total_cost,
            "success_rate": stats.success_rate,
            "top_features": [],
            "has_data": stats.total_requests > 0,
        }
    )
